client/init.js

```diff
@@ -87,17 +88,45 @@ addEventListener('load', function() {
 
 	function startWatchingPosition() {
 		navigator.geolocation.watchPosition(function(position) {
-			try {
-				comms.send({
-					event: 'friend-updated',
-					position: position
-				})
-			} catch (e) {
-				console.error(e)
-			}
+			sendMessage({
+				event: 'friend-updated',
+				position: position
+			})
 		})
 	}
 
+	function setupHiddenEvent(){
```

Comment:
I don't know how hard this would be, but what if the Visibility object took the comms object as an argument, and did all of its event firing in there?